differentiate between subgraph and supergraph in getting-started (#29)
apollographql/rover-init-starters: start-with-typescript/getting-started.md

## Apollo Federation components

This project gets you started with building a federated architecture called a _supergraph_. This architecture lets different teams independently develop and deploy parts of the supergraph while maintaining a unified experience for clients.

> A supergraph contains the _router_, and one or more _subgraphs_.

Reviewer: Should we say *the* router, or a *a* router? (Unsure)

Author: Ahh good catch. I think it's _a_ router.

Author: ```suggestion
A supergraph contains a _router_, and one or more _subgraphs_.
```

**📓 Note:** This graph is using [Apollo Server](https://github.com/apollographql/apollo-server)—an open source server library that is quick and easy to set up, giving you a way to build a production-ready, self-documenting GraphQL API.

# Make your first request

> ## To the subgraph server

Author: ```suggestion
## To the subgraph server (http://localhost:4001)
```
Would adding the URL here help differentiate?

Reviewer: Yes! I think that'd be helpful!

5. Click `► GetProducts` to run the request. You'll get a response back with data for the product's id, name, and description; exactly the properties you asked for in the query! 🎉

When you want to test your subgraph in isolation, visit http://localhost:4001.

### To the supergraph

> A supergraph is composed of one or more subgraphs (right now, you're only starting with one), and can be accessed with a single endpoint.

Author: Feel like we almost need to "sell" this more here at the start. At the moment, because we only have one subgraph, it feels unnecessary to run rover dev. We have this challenge with our courses all the time 😅

Author: We can remove this now with the addition of the federation section above?

Reviewer: I agree. I do like the callout that you're only working with one subgraph fow now, but maybe that's clear enough in the federation section already? What do you think?

1. In a _new_ terminal window, run the `rover dev` command provided in the output of `rover init` under **Next steps**. The `dev` command starts a local development session with a supergraph.
2. In the browser, go to http://localhost:4000, where `rover dev` is running and gives you access to Sandbox. Make sure you still have the subgraph server running from the previous section.
3. In Sandbox, paste the same GraphQL query in the **Operation** section:

```
query GetProducts {
  products {
    id
    name
    description
  }
}
```

> 4. Click `► GetProducts` to run the request. You'll get the same response back as before.

Author: Or maybe we put the explanation here? Because right now this leaves me feeling like... why'd I do all that

5. Click `► GetProducts` to run the request. You'll get a response back with data for the product's id, name, and description; exactly the properties you asked for in the query! 🎉

## To the supergraph (http://localhost:4000)

1. In a _new_ terminal window, run the `rover dev` command provided in the output of `rover init` under **Next steps**. The `dev` command starts a local development session with the router.

> 2. In the browser, go to http://localhost:4000, where `rover dev` is running, and gives you access to another Sandbox. Make sure you still have the subgraph server running from the previous section.

Author: ```suggestion
2. In the browser, go to http://localhost:4000, where `rover dev` is running. You'll have access to another Sandbox. Make sure you still have the subgraph server running from the previous section.
```

3. In Sandbox, paste the same GraphQL query in the **Operation** section:

```
query GetProducts {
  products {
    id
    name
    description
  }
}
```

4. Click `► GetProducts` to run the request. You'll get the same response back as before, but this time, the request was handled by the router.

> ℹ️ **Tip:** To learn more about which port to work with, check out the section titled [Run test requests in Sandbox](#run-test-requests-in-sandbox).

Reviewer: Added this here as a tip to close out the section—this way we surface the info early but keep it where it's most relevant. What do you think, @mabuyo?

Author: Yes, I think this helps!!

5. Click `► GetProducts` to run the request. You'll get a response back with data for the product's id, name, and description; exactly the properties you asked for in the query! 🎉

When you want to test your subgraph in isolation, visit http://localhost:4001.

> ## To the supergraph

Author: ```suggestion
## To the supergraph (http://localhost:4000)
```

A supergraph is composed of one or more subgraphs (right now, you're only starting with one), and can be accessed with a single endpoint.

1. In a _new_ terminal window, run the `rover dev` command provided in the output of `rover init` under **Next steps**. The `dev` command starts a local development session with the router.
2. In the browser, go to http://localhost:4000, where `rover dev` is running, and gives you access to Sandbox. Make sure you still have the subgraph server running from the previous section.
3. In Sandbox, paste the same GraphQL query in the **Operation** section:

```
query GetProducts {
  products {
    id
    name
    description
  }
}
```

> 4. Click `► GetProducts` to run the request. You'll get the same response back as before.
> 
> When you want to test your supergraph, visit http://localhost:4000. `rover dev` is how you test and validate your supergraph. As more services/subgraphs get added to your graph, you'll need to validate that one subgraph server's changes work correctly with the rest of your supergraph.

Author: ```suggestion
4. Click `► GetProducts` to run the request. You'll get the same response back as before, but this time, the request was handled by the router. 

As you continue to add more subgraphs to the supergraph, you'll need to validate that one subgraph's changes work correctly with the rest of your supergraph.
```

5. Click `► GetProducts` to run the request. You'll get a response back with data for the product's id, name, and description; exactly the properties you asked for in the query! 🎉

## To the supergraph (http://localhost:4000)

1. In a _new_ terminal window, run the `rover dev` command provided in the output of `rover init` under **Next steps**. The `dev` command starts a local development session with the router.

> 2. In the browser, go to http://localhost:4000, where `rover dev` is running, and gives you access to another Sandbox. Make sure you still have the subgraph server running from the previous section.

Reviewer 2: We may want to draw more attention to this point (maybe with either the `Note` or `Tip` tag?

```suggestion
2. In the browser, go to http://localhost:4000, where `rover dev` is running, and gives you access to another Sandbox. 
**📓 Note:** Make sure you still have the subgraph server running from the previous section.
```

Reviewer 2: nvm! I see the callout below

3. In Sandbox, paste the same GraphQL query in the **Operation** section:

```
query GetProducts {
  products {
    id
    name
    description
  }
}
```

4. Click `► GetProducts` to run the request. You'll get the same response back as before, but this time, the request was handled by the router.

## Choosing a port to work with

> As you start building and updating your schema, Apollo Sandbox lets you validate your changes by testing requests and examining the corresponding server responses.

Author: ```suggestion
As you start building your schema, you can use Apollo Sandbox to send requests to your subgraph or supergraph.
```
Simplifying, and bringing more context to the next two paragraphs.

When you want to test your subgraph server in isolation, use the Sandbox running at http://localhost:4001, started by the `npm run dev` command. This is recommended when you're just starting out, or when you want to focus on a specific subgraph server.

When you want to test the supergraph as a whole, use the Sandbox running at http://localhost:4000, started by the `rover dev` command. This is recommended when you have more than one subgraph in your supergraph.

> 📓 **Note:** Please keep in mind that if you choose to work with `rover dev` and `localhost:4000`, you'll need to start the subgraph server _first_ by running `npm ci` and `npm run dev`—otherwise, you'll encounter errors when running requests in Sandbox.

Author: ```suggestion
📓 **Note:** If you are using `rover dev` and `localhost:4000`, you'll need to start the subgraph server _first_ by running `npm ci` and `npm run dev`—otherwise, you'll encounter errors when running requests in Sandbox.
```
I feel so bad removing your polite words @ana-demags 😭 But I do think the less words we have and the more direct we are, the better for readability and for those just skimming.

Reviewer: Aw noooo, please don't feel bad @mabuyo! This suggestion is excellent and very polite—the more time we can save readers, the better. 🥹❤️

As you update your schema, Apollo Sandbox lets you validate your changes by testing requests and examining the actual server responses.

When you want to test your subgraph server in isolation, use the Sandbox running at http://localhost:4001, started by the `npm run dev` command. We recommend this when you're just starting out, or when you want to focus on a specific subgraph server.

When you want to test the supergraph as a whole, use the Sandbox running at http://localhost:4000, started by the `rover dev` command. We recommend this any time you have more than one subgraph in your supergraph.

> 

Author: Open to wording here, but wdyt @ana-demags

Author: This is under the "Run test requests in Sandbox" section, which is when it will be most relevant

Reviewer: I agree that this is the place where it makes the most sense—I'm a bit torn because of how far down it is (considering how quickly they'll run into this `localhost` fork in the terminal after running `init`). I'll give it another couple of reads to see if I can think of something, but I think this is in a very shippable state! 💟

Reviewer 2: I could also see this working under the `Make your first request` section if we want to just introduce this concept of "you have commands to run both. here is when you should run either/or. here is how you run them."

I think we also lose a little context that they need to be running their local server to query it when running `rover dev`, otherwise they will see errors.  It builds on top of running the router and is a way for them to layer other services for testing on top of any services they are running locally.
